src/http/router/route.zig

```diff
@@ -0,0 +1,206 @@
+        }
+
+        pub fn put(self: Self, handler_fn: HandlerFn) Self {
+            return inner_route(.PUT, self, handler_fn);
+        }
+
+        pub fn delete(self: Self, handler_fn: HandlerFn) Self {
+            return inner_route(.DELETE, self, handler_fn);
+        }
+
+        pub fn connect(self: Self, handler_fn: HandlerFn) Self {
+            return inner_route(.CONNECT, self, handler_fn);
+        }
+
+        pub fn options(self: Self, handler_fn: HandlerFn) Self {
+            return inner_route(.OPTIONS, self, handler_fn);
+        }
+
+        pub fn trace(self: Self, handler_fn: HandlerFn) Self {
+            return inner_route(.TRACE, self, handler_fn);
+        }
+
+        pub fn patch(self: Self, handler_fn: HandlerFn) Self {
+            return inner_route(.PATCH, self, handler_fn);
+        }
+
+        /// Define a GET handler to serve an embedded file.
+        pub fn serve_embedded_file(
+            self: *const Self,
+            comptime mime: ?Mime,
+            comptime bytes: []const u8,
+        ) Self {
+            return self.get(struct {
+                fn handler_fn(ctx: *Context) !void {
+                    const cache_control: []const u8 = if (comptime builtin.mode == .Debug)
+                        "no-cache"
+                    else
+                        comptime std.fmt.comptimePrint(
+                            "max-age={d}",
+                            .{std.time.s_per_day * 30},
+                        );
+
+                    ctx.response.headers.putAssumeCapacity("Cache-Control", cache_control);
+
+                    // If our static item is greater than 1KB,
+                    // it might be more beneficial to using caching.
+                    if (comptime bytes.len > 1024) {
+                        @setEvalBranchQuota(1_000_000);
+                        const etag = comptime std.fmt.comptimePrint("\"{d}\"", .{std.hash.Wyhash.hash(0, bytes)});
+                        ctx.response.headers.putAssumeCapacity("ETag", etag[0..]);
+
+                        if (ctx.request.headers.get("If-None-Match")) |match| {
+                            if (std.mem.eql(u8, etag, match)) {
+                                try ctx.respond(.{
+                                    .status = .@"Not Modified",
+                                    .mime = Mime.HTML,
+                                    .body = "",
+                                });
+
+                                return;
+                            }
+                        }
+                    }
+
+                    try ctx.respond(.{
+                        .status = .OK,
+                        .mime = mime,
+                        .body = bytes,
+                    });
+                }
+            }.handler_fn);
+        }
+
+        /// Define a GET handler to serve an entire directory.
+        pub fn serve_fs_dir(comptime _self: *const Self, comptime dir_path: []const u8) Self {
```

Comment:
`_self` ->`self`. We tend to use the `_name` syntax when I'm importing a type that requires comptime parameters to become fully qualified, like `_Context` -> `Context`.

Reply:
Oh, I didn't notice that `_Name` was only for importing types with comptime parameters. My personal naming convention is to add a _ for any name collision. Here, I think it's because I tried a lot of things with this function and forgot to rename it back to `self` instead of `_self` after all my tests. Nice catch!